source/Irrlicht/CImageLoaderBMP.cpp

```diff
@@ -107,26 +113,29 @@ void CImageLoaderBMP::decompress8BitRLE(u8*& bmpData, s32 size, s32 width, s32 h
 		}
 	}
 
+exit:
 	delete [] bmpData;
 	bmpData = newBmp;
 }
 
+#define SHIFTED_COUNT(_count, _shift) ((_count) / 2 + (((_shift) == 0 || (_count) % 2 == 1) ? 1 : 0))

```

Comment:
What does it do?
And, why a macro? Wouldn’t inline static function be equally suitable and more readable?

Reply:
the code operates on 4 bits, this computes how many bytes it will touch given the current state

Reply:
>Wouldn’t inline static function be equally suitable and more readable?

Or a lambda in the function.